src/evaluator/program.cpp

```diff
@@ -502,6 +593,69 @@ void Call::display(std::ostream& stream, size_t depth) const {
 
 bool Call::can_evaluate_to_function() const { return true; }
 
+Cond::Cond(
+    std::unique_ptr<Expression> condition,
+    std::unique_ptr<Expression> then,
+    std::unique_ptr<Expression> otherwise)
+    : Expression(), condition(std::move(condition)), then(std::move(then)), otherwise(std::move(otherwise)) {}
+
+std::unique_ptr<Cond> Cond::parse(std::shared_ptr<ast::List> arguments) {
+    auto cons = arguments->to_cons();
+
+    if (!cons) {
+        throw std::runtime_error("`cond` takes 2 or 3 arguments, provided 0");
+    }
+
+    auto condition = Expression::from_element(cons->left);
+
+    cons = cons->right->to_cons();
+
+    if (!cons) {
+        throw std::runtime_error("`cond` takes 2 or 3 arguments, provided 1");
+    }
+
+    auto then = Expression::from_element(cons->left);
+
+    cons = cons->right->to_cons();
+
+    std::unique_ptr<Expression> otherwise;
+    if (!cons) {
+        otherwise = Expression::from_element(std::make_shared<Null>(Null(arguments->span)));
+    }
+    else {
+        otherwise = Expression::from_element(cons->left);
+    }
+
+    auto cond = Cond(std::move(condition), std::move(then), std::move(otherwise));
+    return std::make_unique<Cond>(std::move(cond));
+}
+
+std::shared_ptr<Element> Cond::evaluate() const {
+    throw std::runtime_error("Not implemented");
+}
+
+void Cond::display(std::ostream& stream, size_t depth) const {
+    stream << "Cond [\n";
+
+    stream << Depth(depth + 1) << "Condition: \n" << Depth(depth+2);
+    this->condition->display(stream, depth+2);
+    stream << "\n";
+
+    stream << Depth(depth + 1) << "Then: \n" << Depth(depth+2);
+    this->then->display(stream, depth+2);
+    stream << "\n";
+    
+    stream << Depth(depth + 1) << "Otherwise: \n" << Depth(depth+2);
+    this->otherwise->display(stream, depth+2);
+    stream << "\n";
+
+    stream << Depth(depth) << ']';
+}
+
+bool Cond::can_evaluate_to_function() const { 
+    return this->then->can_evaluate_to_function() && this->otherwise->can_evaluate_to_function();
```

Comment:
```suggestion
    return this->then->can_evaluate_to_function() || this->otherwise->can_evaluate_to_function();
```

если одна ветка может вычислиться в функцию, значит и весь cond может